src/Compilers/Core/CodeAnalysisTest/DefaultAnalyzerAssemblyLoaderTests.cs

```diff
@@ -204,44 +318,105 @@ public void AssemblyLoading()
             });
         }
 
-        [ConditionalFact(typeof(CoreClrOnly))]
-        public void AssemblyLoading_AssemblyLocationNotAdded()
+        /// <summary>
+        /// The loaders should not actually look at the contents of the disk until a <see cref="AnalyzerAssemblyLoader.LoadFromPath(string)"/>
+        /// call has occurred. This is historical behavior that doesn't have a clear reason for existing. There
+        /// is strong suspicion it's to delay loading of analyzers until absolutely necessary. As such we're
+        /// enshrining the behavior here so it is not _accidentally_ changed.
+        /// </summary>
+        [Theory]
+        [CombinatorialData]
+        public void AssemblyLoading_OverwriteBeforeLoad(bool shadowLoad)
         {
-            Run(static (DefaultAnalyzerAssemblyLoader loader, AssemblyLoadTestFixture testFixture) =>
+            Run(shadowLoad, static (DefaultAnalyzerAssemblyLoader loader, AssemblyLoadTestFixture testFixture) =>
+            {
+                loader.AddDependencyLocation(testFixture.Delta1.Path);
+                testFixture.Delta1.WriteAllBytes(testFixture.Delta2.ReadAllBytes());
+                var assembly = loader.LoadFromPath(testFixture.Delta1.Path);
+
+                var name = AssemblyName.GetAssemblyName(testFixture.Delta2.Path);
+                Assert.Equal(name.FullName, assembly.GetName().FullName);
```

Comment:
Would it make sense to use `VerifyAssemblies()` here instead, to check that the assembly is in the shadow directory, and perhaps other details?
```
VerifyAssemblies(loader, new[] { assembly }, ("Delta", "2.0.0.0", testFixture.Delta2.Path));
```